front-end/src/components/Login/index.js

```diff
@@ -40,53 +44,113 @@ class Login extends Component {
+  error = React.createRef();
+
   render () {
+    const {lang} = this.props;
     return (
-      <div className="login" ref="login">
-        <div className="login__dialog">
-          <form action="#" className="login__form" onSubmit={this.handleSubmit}>
-            <label htmlFor="username">Логин</label>
-            <input type="text" name="username" id="username" placeholder="введите имя пользователя (Admin)"
-              value={this.state.username} onChange={this.handleChange} required/>
-            <label htmlFor="password">Пароль</label>
-            <input type="password" name="password" id="password" placeholder="введите пароль (1234)"
-              value={this.state.password} onChange={this.handleChange} required/>
-            <input type="submit" name="" value="Войти"/>
-          </form>
-          <span className="login__data-error">неверный логин или пароль</span>
+      <IntlProvider locale={lang} messages={messages[lang]}>
+        <div className="login" ref="login">
+          <div className="login__locale"><Locale/></div>
+          <div className="login__dialog">
+            <form action="#" className="login__form" ref={form => (this.form = form)} onSubmit={this.handleSubmit}>
+              <label htmlFor="username" className="login__label">
+                <FormattedMessage id="login_login" defaultMessage="Логин"/>
+              </label>
+              <input type="text" className="login__input" ref={username => (this.username = username)}
+                name="username" id="username" value={this.state.username} onChange={this.handleChange} required/>
+              <label htmlFor="password" className="login__label">
+                <FormattedMessage id="login_password" defaultMessage="Пароль"/>
+              </label>
+              <input type="password" className="login__input" ref={password => (this.password = password)}
+                name="password" id="password" value={this.state.password} onChange={this.handleChange} required/>
+              <input type="submit" className="login__input" ref={submitBtn => (this.submitBtn = submitBtn)}
+                name="" value="Войти"/>
+            </form>
+            <span className="login__data-error" ref={error => (this.error = error)}>
+              <FormattedMessage id="login_wrong-data" defaultMessage="неверный логин или пароль"/>
+            </span>
+          </div>
         </div>
-      </div>
+      </IntlProvider>
     );
   }
 
-    componentDidMount = () => {
-      let form = $('form');
-      let submitButton = $('input[type="submit"]');
+  componentDidMount = () => {
+    let form = this.form;
+    let username = this.username;
+    let password = this.password;
+    let submitBtn = this.submitBtn;
+    let error = this.error;
 
-      form.on('keydown', function (event) {
-        if (event.keyCode === 13) {
-          submitButton.css('transform', 'scale(.99)');
-        }
-      });
+    if (localStorage.lang === 'en') {
```

Comment:
Here is a good example. Imagine you have 10 languages and you have to create 10 conditions to set these placeholders. Such code would be very hard to maintain.

Reply:
agree